plugin/src/rules/require_await.rs

```diff
@@ -0,0 +1,243 @@
+use std::sync::Arc;
+
+use tree_sitter_lint::{rule, violation, Rule};
+
+use crate::{
+    ast_helpers::{is_async_function, is_for_of_await},
+    kind::{GeneratorFunction, GeneratorFunctionDeclaration},
+    utils::ast_utils,
+};
+
+fn capitalize_first_letter(text: &str) -> String {
+    format!("{}{}", text[0..1].to_uppercase(), &text[1..])
+}
+
+pub fn require_await_rule() -> Arc<dyn Rule> {
+    rule! {
+        name => "require-await",
+        languages => [Javascript],
+        messages => [
+            missing_await => "{{name}} has no 'await' expression.",
+        ],
+        state => {
+            [per-file-run]
+            scope_info: Vec<bool>,
```

Comment:
The ESLint version uses a recursive data structure for this but I'm not sure I see the upside of that vs a "stack" so I did it as a stack